### PR TITLE
Actually Perform the Cleanup of Checks

### LINKED ISSUE #393: K8up does not clean up check jobs
## Describe the bug

I set the env variable `BACKUP_GLOBALKEEPJOBS`.
K8up cleans then `prune` and `backup` jobs, but not `check` jobs.

## Expected behavior

If `BACKUP_GLOBALKEEPJOBS` is set, k8up will cleanup `backup`, `check` and `prune` jobs.

## Environment (please complete the following information):

- Image Version: e.g. v1.0.2
- K8s Version: v1.20.4+k3s1
- K8s Distribution: k3s

### PR DESCRIPTION
**Pull request by cimnine**
## Summary

It was discovered that the cleanup of Checks was not implemented. This PR adds this missing functionality.

Closes #393

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [x] Update the documentation.
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

**Comment by cimnine** [comment]
@tobru I've requested your review as I updated the GoLand instructions in the docs.

**Comment by cimnine** [comment]
@ccremer I think codeclimate's duplicate code issue is not worth the effort, because the core part of the deduplication logic is already extracted to a common place. The part which is supposedly duplicated is the code that prepares the data structures to leverage the deduplication function. Because of Go's type system, it looks like an exact copy, yet it isn't.

WDYT?

**Comment by cimnine** [comment]
It strikes me as weird that the diff-coverage did not increase at all, since I've added tests that cover code not tested before 🤔  Is integrationtest coverage perhaps not reported correctly?

**Review by tobru — APPROVED** [review]

File: controllers/check_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		}
+
+		c.T().Logf("%d deleted Checks found", amountOfDeletedItems)
+		done = amountOfDeletedItems == expectedDeletes
+
+		return
+	})
+}
+
+func (c *CheckTestSuite) whenJobCallbackIsInvoked(checkName string) {
```
Comment (ccremer) [inline controllers/check_integration_test.go:102]: I recommend to rename this to `ts` to be consistent with other integration tests and give same dev experience.
```suggestion
func (ts *CheckTestSuite) whenJobCallbackIsInvoked(checkName string) {
```

File: controllers/check_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	checkNSName := types.NamespacedName{Name: checkName, Namespace: c.NS}
+
+	check := &batchv1.Job{}
+	err := c.Client.Get(c.Ctx, checkNSName, check)
+	c.Require().NoError(err)
```
Comment (ccremer) [inline controllers/check_integration_test.go:107]: there should be a `fetchResource()` function from EnvTestSuite already, saving you from some DRY

File: executor/check.go [hunk 3]
```diff
@@ -48,12 +55,13 @@ func (c *CheckExecutor) Execute() error {
 	}
 	checkJob.GetLabels()[job.K8upExclusive] = "true"
 
-	return c.startCheck(checkObject, checkJob)
+	return c.startCheck(checkJob, checkObject)
 }
 
-func (c *CheckExecutor) startCheck(checkObject *k8upv1alpha1.Check, checkJob *batchv1.Job) error {
+func (c *CheckExecutor) startCheck(checkJob *batchv1.Job, checkObject *k8upv1alpha1.Check) error {
 	c.check = checkObject
```
Comment (ccremer) [inline executor/check.go:62]: Line 62 assigns does `c.check = checkObject`, why can't we do that within Execute() already? that would save us a parameter

**Comment by ccremer** [comment]
> Because of Go's type system, it looks like an exact copy, yet it isn't.

In CodeClimate UI, you can mark this issue as invalid or wontfix with your reasoning. then it becomes green and should be saved for later PRs as well.

**Comment by ccremer** [comment]
> It strikes me as weird that the diff-coverage did not increase at all, since I've added tests that cover code not tested before :thinking: Is integrationtest coverage perhaps not reported correctly?

I'd recommend to run `make integration-test` once in current master branch and then compare with your PR. you could also run all integration tests with GoLand with Coverage enabled to see which parts of the code gets executed and which not. It's possible that some other integration test is going through the same code locations already, but it still shouldn't be 0% I guess...

Be aware that code coverage in Goland is not calculated the same as gotest but it may give some clue.

**Comment by cimnine** [comment]
> In CodeClimate UI, you can mark this issue as invalid or wontfix with your reasoning. then it becomes green and should be saved for later PRs as well.

I'm afraid, I don't have these privileges.

**Comment by cimnine** [comment]
> I'd recommend to run make integration-test once in current master branch and then compare with your PR.

It increases form 30% to 38%.

> you could also run all integration tests with GoLand with Coverage enabled

For `controllers/check.go`, it increases from 0% to 60%.

I think that's a good-enough ratio :)

File: docs/modules/ROOT/pages/explanations/ide.adoc [hunk 4]
```diff
@@ -9,18 +9,18 @@ You can basically just open the root directory in GoLand and you'll be mostly re
...
 
-Now try to re-run the test.
-It should now succeed.
-You will see test results in your IDE and be able to use the integrated debugger as well.
+You can use all the usual shortcuts to quickly run a test (e.g. kbd:[Ctrl+Shift+D] to debug a specific test method).
+If it doesn't work, check the following:
+
+. Check that the _Custom tags_ field in _Preferences_ → _Go_ → _Build Tags & Vendoring_ contains to term `integration`.
+. Make sure the _Working directory_ of your test's _Run Configuration_ points to the directory where your test lies,
+  e.g. `/home/your_user/projects/k8up/controllers`.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/explanations/ide.adoc:22]: We have a one-sentence-per-line styleguide, so this should be all one line
```suggestion
. Make sure the _Working directory_ of your test's _Run Configuration_ points to the directory where your test lies, e.g. `/home/your_user/projects/k8up/controllers`.
```

**Comment by ccremer** [comment]
The coverage is indeed strange...
well, what matters is that we have a working test case :)

**Review by ccremer — APPROVED** [review]

**Pull request merged**